json/src/ser.rs

```diff
@@ -504,6 +508,28 @@ pub fn escape_bytes<W>(wr: &mut W, bytes: &[u8]) -> Result<()>
             b'\n' => b"\\n",
             b'\r' => b"\\r",
             b'\t' => b"\\t",
+            b'\x00' ... b'\x1F' | b'\x7F' => {
+                if start < i {
+                    try!(wr.write_all(&bytes[start..i]));
+                }
+
+                try!(write!(wr,"\\u{:04X}", *byte));
+
+                start = i + 1;
+
+                continue;
+            },
+            b'\x80' ... b'\x9F' if last_byte_was_c2 => {
```

Comment:
I think instead of trying to implement half of a UTF8 decoder here, it would be better to have [`escape_str`](https://github.com/serde-rs/json/blob/7bc9b0a98ec65b90f8c4600d5966d1ca0679b089/json/src/ser.rs#L532) not implemented in terms of `escape_bytes` but instead use [`.chars()`](https://doc.rust-lang.org/std/primitive.str.html#method.chars) to walk through the UTF8 string.